content/en/blog/_posts/2022-12-6-provision-volumes-from-cross-namespace-snapshots-alpha.md

```diff
@@ -0,0 +1,133 @@
+ To solve this problem,Kubernetes v1.26 includes a new API field called `dataSourceRef2`.
+
+## How it works
+
+Once the csi-provisioner finds a VolumeSnapshot is specified with non-empty namespace as dataSourceRef2, it checks all ReferenceGrants in　PersistentVolumeClaim.spec.dataSourceRef2.namespace to see if access to the snapshot is allowed. If it is allowed, the csi-provisioner provisions a volume from the snapshot.  
+
+## Trying it out
+
+The following things are required to use cross namespace volume provisioning:
+
+* Enable the `CrossNamespaceVolumeDataSource` feature gate
+* Install a CRD for the specific `VolumeSnapShot controller`
+* Install the `VolumeSnapShot controller` itself
+* Install the `External Provisioner controller` itself
+* Install the `Container Stroge Interface Driver` itself
+
+## Putting it all together
+
+To see how this works, you can install the sample and try it out.
+This sample do to create PVC in ns1 namespace from VolumeSnapshot in default namespace by using csi-hostpath driver.
+
+1. Deploy VolumeSnapshot controller.
+
+   ```terminal
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshotclasses.yaml
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshotcontents.yaml
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshots.yaml
+
+
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/rbac-snapshot-controller.yaml
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml
+   ```
+
+2. Deploy External Provisioner controller.
+
+   ```terminal
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-provisioner/master/deploy/kubernetes/deployment.yaml 
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-provisioner/master/deploy/kubernetes/rbac.yaml
+   ```
+
+3. Deploy csi-hostpath driver.
+
+   ```terminal
+   cd /tmp
+   git clone --depth=1 https://github.com/kubernetes-csi/csi-driver-host-path.git
+   cd csi-driver-host-path
+   ./deploy/kubernetes-latest/deploy.sh
+   ```
+
+4. Deploy ReferenceGrants CRD
+  
+   ```terminal
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/main/config/ crd/experimental/gateway.networking.k8s.io_referencegrants.yaml
+   ```
+
+5. Create StorageClass, PVC, and VolumeSnapshot by the examples in the csi-hostpath repo
+
+    ```terminal
```

Comment:
(nit)
I'd highlight this as shell. Check that the indent is correct (should be 3 spaces).